src/cpp/rtps/builtin/discovery/participant/PDPSimple.cpp

```diff
@@ -273,6 +278,7 @@ bool PDPSimple::createPDPEndpoints()
         mp_PDPReaderHistory = nullptr;
         delete mp_listener;
         mp_listener = nullptr;
+        reader_payload_pool_->release_history(reader_pool_cfg, true);
```

Comment:
`reader_payload_pool` should be released on the registry too.
PDP destructor only does that if `mp_PDPReaderHistory` is not null (which we have just set)